File: sssapril/vov/client/src/types/resource.ts
/**
 * 资源相关类型定义模块
 *
 * 定义资源(Resource)和标签(Tag)相关的TypeScript类型。
 */

import { ContentType, ResourceType, Scope } from './common';

/**
 * 资源信息
 *
 * 资料是项目/群聊的参考材料。
 */
export interface Resource {
  /** 资源ID */
  id: string;
  /** 所属项目ID */
  project_id: string;
  /** 所属群聊ID（NULL表示全局资源） */
  group_id: string | null;
  /** 资源标题 */
  title: string;
  /** 资源内容（Markdown） */
  content: string;
  /** 内容类型 */
  content_type: ContentType;
  /** 资源类型 */
  type: ResourceType;
  /** 标签列表 */
  tags: string[];
  /** 是否必读 */
  is_required: boolean;
  /** 创建者 */
  created_by: string;
  /** 创建时间 */
  created_at: string;
  /** 更新时间 */
  updated_at: string;
}

/**
 * 创建资源请求参数
 */
export interface CreateResourceRequest {
  /** 资源标题 */
  title: string;
  /** 资源内容 */
  content: string;
  /** 内容类型 */
  content_type?: ContentType;
  /** 资源类型 */
  type?: ResourceType;
  /** 标签 */
  tags?: string[];
  /** 是否必读 */
  is_required?: boolean;
  /** 所属项目ID */
  project_id: string;
  /** 所属群聊ID（不传表示全局资源） */
  group_id?: string;
  /** 创建者 */
  created_by?: string;
}

/**
 * 更新资源请求参数
 */
export interface UpdateResourceRequest {
  /** 资源标题 */
  title?: string;
  /** 资源内容 */
  content?: string;
  /** 资源类型 */
  type?: ResourceType;
  /** 标签 */
  tags?: string[];
  /** 是否必读 */
  is_required?: boolean;
}

/**
 * 项目标签
 *
 * 每个项目独立管理自己的标签体系。
 */
export interface Tag {
  /** 标签ID */
  id: string;
  /** 所属项目ID */
  project_id: string;
  /** 标签名称 */
  name: string;
  /** 标签说明 */
  description: string | null;
  /** 建议的格式/模板 */
  suggested_template: string | null;
  /** 标签颜色 */
  color: string | null;
  /** 创建时间 */
  created_at: string;
  /** 更新时间 */
  updated_at: string;
}

/**
 * 创建标签请求参数
 */
export interface CreateTagRequest {
  /** 标签名称 */
  name: string;
  /** 标签说明 */
  description?: string;
  /** 建议的格式/模板 */
  suggested_template?: string;
  /** 标签颜色 */
  color?: string;
}

/**
 * 更新标签请求参数
 */
export interface UpdateTagRequest {
  /** 标签名称 */
  name?: string;
  /** 标签说明 */
  description?: string;
  /** 建议的格式/模板 */
  suggested_template?: string;
  /** 标签颜色 */
  color?: string;
}

File: sssapril/vov/server/app/schemas/resource.py
"""
资源Schema模块

定义资源(Resource)相关的Pydantic schemas。
"""

from typing import Optional, List, Dict, Any
from datetime import datetime
from pydantic import BaseModel, Field

from .common import TimestampMixin


class ResourceBase(BaseModel):
    """资源基础Schema"""
    title: str = Field(..., min_length=1, max_length=255, description="资源标题")
    content: str = Field(..., description="资源内容（Markdown格式）")
    content_type: str = Field("markdown", description="内容类型")
    type: str = Field("note", description="资源类型: note/reference/guideline/rule/custom/map")
    tags: List[str] = Field(default_factory=list, description="标签列表")
    is_required: bool = Field(False, description="是否必读")
    created_by: str = Field("user", description="创建者: user 或 agent_id")


class ResourceCreate(ResourceBase):
    """创建资源请求"""
    project_id: str = Field(..., description="所属项目ID")
    group_id: Optional[str] = Field(None, description="所属群聊ID，NULL表示全局资源")


class ResourceUpdate(BaseModel):
    """更新资源请求"""
    title: Optional[str] = Field(None, min_length=1, max_length=255, description="资源标题")
    content: Optional[str] = Field(None, description="资源内容")
    type: Optional[str] = Field(None, description="资源类型")
    tags: Optional[List[str]] = Field(None, description="标签列表")
    is_required: Optional[bool] = Field(None, description="是否必读")


class ResourceResponse(ResourceBase, TimestampMixin):
    """资源响应"""
    id: str = Field(..., description="资源ID")
    project_id: str = Field(..., description="项目ID")
    group_id: Optional[str] = Field(None, description="群聊ID")
    task_id: Optional[str] = Field(None, description="关联任务ID")
    parent_id: Optional[str] = Field(None, description="父资源ID（文件夹）")
    is_folder: bool = Field(False, description="是否为文件夹")

    class Config:
        from_attributes = True
